docs/DeveloperGuide.md

```diff
@@ -1,38 +1,181 @@
+
+4. TransactionList: A data structure to store and manage transactions.
+
+5. Transaction: Represents individual transactions.
+
+6. UI: Handles user interface and messaging.
+
+#### II. Component-Level Design
+1. Parser
+
+The Parser class identifies the "view" command and extracts the taskIndex (transaction index) from the user's input.
+
+2. ViewCommand
+
+The ViewCommand object is created by the Parser. It encapsulates the user's request to view a specific transaction. This object is passed to the Nuscents class for execution.
+
+3. Nuscents
+
+In the Nuscents class, the execute() method of the ViewCommand is called, and the taskIndex is extracted from the command. It then calls the viewTransaction(taskIndex) method on the TransactionList.
+
+4. TransactionList
+
+The TransactionList contains a list of Transaction objects. The viewTransaction(taskIndex) method retrieves the specific Transaction object based on the taskIndex.
+
+5. Transaction
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The Transaction class represents an individual transaction, and it contains all the relevant details of a transaction.
+
+6. UI
+
+The UI class displays the transaction details using the UI.showTransactionViewMessage method, which receives the Transaction object as input and presents the detailed transaction information to the user.
+
+#### III. Alternatives Considered
+While the current design is deemed suitable for our application, we did consider alternative approaches, such as 
+integrating the view transaction functionality directly within the Nuscents class without introducing a ViewCommand. 
+However, we opted for the current design to promote a cleaner separation of concerns and to facilitate future expansions 
+and modifications.
+
+The following sequence diagram shows how the add transaction operation works:
+<img src="images/ViewSequenceDiagram.png" width="600" />
+
+### List transactions feature
+The list transaction feature is facilitated by the `Parser` class which parses user input and creates a new
+`ListCommand` object. The `ListCommand` object will get all the transactions in the `TransactionList` and display them 
+to the user.
+
+Given below is the example usage scenario and how the list transaction mechanism behaves at each step.
+
+Step 1. The user launches the application. The `TransactionList` will be initialized with the transactions stored in 
+the `nuscents.txt` file. If the file is empty or does not exist, the `TransactionList` will be empty. 
+
+Step 2. The user executes `list` command to list the transactions. The `list` command calls `ListCommand#execute()`, 
+which gets the transactions from the `TransactionList` and displays them to the user.
+
+The following sequence diagram shows how the list transaction operation works:
+
+<img src="images/ListTransactionSequenceDiagram.png" width="600" />
```

Comment:
![Uploading Screenshot 2023 -11-02 at 5.39.34 PM.jpg…]()
Are the duplicate classes at the bottom of the diagram required? Could make it confusing in terms of sequence flow.